lib/Utils/Checker/Rules/RequestedClaimsRule.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace SimpleSAML\Module\oidc\Utils\Checker\Rules;
+
+use League\OAuth2\Server\Repositories\ClientRepositoryInterface;
+use OpenIDConnectServer\ClaimExtractor;
+use Psr\Http\Message\ServerRequestInterface;
+use SimpleSAML\Module\oidc\Entity\Interfaces\ClientEntityInterface;
+use SimpleSAML\Module\oidc\Server\Exceptions\OidcServerException;
+use SimpleSAML\Module\oidc\Utils\Checker\Interfaces\ResultBagInterface;
+use SimpleSAML\Module\oidc\Utils\Checker\Interfaces\ResultInterface;
+use SimpleSAML\Module\oidc\Utils\Checker\Result;
+
+class RequestedClaimsRule extends AbstractRule
+{
+    private $clientRepository;
+
+    private $claimExtractor;
+
+    public function __construct(ClientRepositoryInterface $clientRepository, ClaimExtractor $claimExtractor)
+    {
+        $this->clientRepository = $clientRepository;
+        $this->claimExtractor = $claimExtractor;
+    }
+
+
+    public function checkRule(
+        ServerRequestInterface $request,
+        ResultBagInterface $currentResultBag,
+        array $data = [],
+        bool $useFragmentInHttpErrorResponses = false
+    ): ?ResultInterface {
+        $claimsParam = $request->getQueryParams()['claims'] ?? null;
+        $claims = json_decode($claimsParam, true);
+        if (is_null($claims)) {
+            return null;
+        }
+        $clientId = $request->getQueryParams()['client_id'] ?? $request->getServerParams()['PHP_AUTH_USER'] ?? null;
```

Comment:
Suggestion: since this rule depends on client ID, and since there is a lib/Utils/Checker/Rules/ClientIdRule.php defined, maybe we could use predefined result, for example, something like lib/Server/Grants/AuthCodeGrant.php:497. This makes it explicitly dependent on another rule, and makes sure that rule is ran before this one.